### ISSUE TITLE
[Bug] Create table in FTS catalog with s3 warehouse throws DatabaseNotExistException

### ISSUE THREAD

**Issue by JingsongLi** [apache/incubator-paimon#733]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

0.4

### Compute Engine

flink

### Minimal reproduce step

```
Flink SQL> CREATE CATALOG my_catalog WITH (
>   'type'='table-store',
>   'warehouse'='s3://bucket/my-tablestore'
> );
[INFO] Execute statement succeed.

Flink SQL> USE CATALOG my_catalog;
[INFO] Execute statement succeed.

Flink SQL> CREATE TABLE word_count (
>     word STRING PRIMARY KEY NOT ENFORCED,
>     cnt BIGINT
> );
[ERROR] Could not execute SQL statement. Reason:
org.apache.flink.table.catalog.exceptions.DatabaseNotExistException: Database default does not exist in Catalog my_catalog. 
```

Creating the table in the default catalog works though:
```
Flink SQL> use catalog default_catalog;
[INFO] Execute statement succeed.

Flink SQL> CREATE TABLE word_count (
>       word STRING PRIMARY KEY NOT ENFORCED,
>       cnt BIGINT
>  ) WITH (
>    'connector'='table-store',
>    'path'='s3://bucket/my-tablestore',
>    'auto-create'='true'
> );
[INFO] Execute statement succeed.
```

### What doesn't meet your expectations?

[ERROR] Could not execute SQL statement. Reason:
org.apache.flink.table.catalog.exceptions.DatabaseNotExistException: Database default does not exist in Catalog my_catalog. 

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by Alibaba-HZY** [comment]
i want to do this issue

**Comment by SteNicholas** [comment]
@JingsongLi, assign this ticke to you.

**Comment by JingsongLi** [comment]
It looks not an issue now.

**Comment by ngk2009** [comment]
This issue is not fixed in the release paimon-flink-1.18-0.6-20231108.091445-47
